[Security] 9.4.0 release notes (#6029)
elastic/docs-content: release-notes/elastic-security/index.md

* Updates MITRE ATT&CK mappings to `v18.1` [#246770]({{kib-pull}}246770).
* Adds new RBAC subfeatures for rules management, allowing granular control over investigation guide editing, custom highlighted fields, enabling/disabling rules, manual rule runs, and rules management settings independently of read-only rules access [#250131]({{kib-pull}}250131).
* Adds an **Exceptions** RBAC subfeature to the Rules privilege, allowing you to grant exception list management independently of rules read/write access [#245722]({{kib-pull}}245722).
* Adds a dedicated **Alerts** {{kib}} feature privilege, separating alert view and edit access from rules management. Alert read/edit can now be granted independently of rules privileges [#244637]({{kib-pull}}244637).
* Adds a project selector to Security Solution pages including rule creation, rule details, rule management, rule editing, **MITRE ATT&CK® coverage**, prebuilt rule installation, and **Shared exception lists** [#258916]({{kib-pull}}258916).

> * Adds an AI Agent skill for querying Security {{ml}} jobs to support {{anomaly-detect}} questions in AI Agent chat [#254908]({{kib-pull}}254908).

Reviewer: ```suggestion
* Adds an Agent Builder skill for querying Security {{ml}} jobs for {{anomaly-detect}} questions [#254908]({{kib-pull}}254908).
```

* Automatically injects `METADATA _id` into {{esql}} detection rule queries during execution, removing the requirement for you to include it manually. A non-blocking warning is shown when automatic injection is not possible [#254703]({{kib-pull}}254703).

> * Adds AI-assisted detection rule creation to AI Agent, generating {{esql}} rules including name, description, query, tags, severity, risk score, schedule, and MITRE ATT&CK mappings from a natural language description [#247674]({{kib-pull}}247674).

Reviewer: ```suggestion
* Adds an Agent Builder skill for creating detection rules [#247674]({{kib-pull}}247674).
```

* Adds entity resolution CSV upload support, including a backend API and UI [#260006]({{kib-pull}}260006).
* Introduces Entity Threat Hunting on the **Entity Analytics** page, surfacing prioritized investigation leads based on behavioral observations, risk scoring, and temporal state analysis with Agent Builder integration for guided investigation. Adds the **Entity Threat Hunting** leads cards, a **See All** flyout, and a lead provenance flyout [#257610]({{kib-pull}}257610).
* Introduces the Entity Threat Hunting backend infrastructure, including the feature flag, shared types and index templates, observation module interface, Risk and Temporal State modules, behavioral observation modules, entity retrieval and enrichment, API routes, `LeadDataClient`, async lead generation, and an automated email-based entity resolution maintainer [#255272]({{kib-pull}}255272), [#256156]({{kib-pull}}256156), [#256270]({{kib-pull}}256270), [#256628]({{kib-pull}}256628), [#257046]({{kib-pull}}257046), [#257479]({{kib-pull}}257479).
* Allows user and host details flyouts to open while observed data is loading [#252657]({{kib-pull}}252657).

> * Adds an Entity Analytics AI Agent skill for answering risk score and entity risk questions in AI Agent chat [#252400]({{kib-pull}}252400).

Reviewer: ```suggestion
* Adds an Entity Analytics Agent Builder skill for answering risk score and entity risk questions [#252400]({{kib-pull}}252400).
```

* Adds an `alerts_candidate_count` rule execution metric [#259917]({{kib-pull}}259917).
* Adds high-level rule execution info logging [#253992]({{kib-pull}}253992).
* Adds UI for identifying and managing deprecated prebuilt detection rules, including a dismissable callout on the rules table page and a callout on the rule details page with actions to delete or duplicate-and-delete deprecated rules [#259673]({{kib-pull}}259673).
* Updates MITRE ATT&CK mappings to `v18.1` [#246770]({{kib-pull}}246770).
* Adds new RBAC subfeatures for rules management, allowing granular control over investigation guide editing, custom highlighted fields, enabling/disabling rules, manual rule runs, and rules management settings independently of read-only rules access [#250131]({{kib-pull}}250131).
* Adds an **Exceptions** RBAC subfeature to the Rules privilege, allowing you to grant exception list management independently of rules read/write access [#245722]({{kib-pull}}245722).
* Adds a dedicated **Alerts** {{kib}} feature privilege, separating alert view and edit access from rules management. Alert read/edit can now be granted independently of rules privileges [#244637]({{kib-pull}}244637).

> * Adds a project selector to Security Solution pages including rule creation, rule details, rule management, rule editing, **MITRE ATT&CK® coverage**, prebuilt rule installation, and **Shared exception lists** [#258916]({{kib-pull}}258916).

Reviewer 2: This is for serverless only.

```suggestion
```
